test/apps/carto-quadbin-cluster/app.tsx

```diff
@@ -0,0 +1,290 @@
+        title="blending"
+        obj={BLENDING}
+        value={blending}
+        onSelect={setBlending}
+        top={60}
+      />
+      <RangeInput
+        name={'Radius'}
+        bottom={150}
+        min={0}
+        max={100}
+        value={radiusRange}
+        onChange={setRadiusRange}
+        formatLabel={(x: number, index: number) => {
+          const label = index === 0 ? 'min' : 'max';
+          return formatLabel(x, label);
+        }}
+      />
+      <RangeInput
+        name={'Cluster level'}
+        bottom={100}
+        min={1}
+        max={7}
+        value={clusterLevel}
+        onChange={setClusterLevel}
+        formatLabel={(x: number) => formatLabel(x, 'level')}
+      />
+      <RangeInput
+        name={'Opacity'}
+        bottom={50}
+        min={0}
+        max={100}
+        value={opacity}
+        onChange={setOpacity}
+        formatLabel={(x: number) => formatLabel(x, '')}
+      />
+      <RangeInput
+        name={'Line width'}
+        bottom={0}
+        min={0}
+        max={10}
+        value={lineWidth}
+        onChange={setLineWidth}
+        formatLabel={(x: number) => formatLabel(x, 'pixels')}
+      />
+    </>
+  );
+}
+
+export function renderToDOM(container: HTMLElement) {
+  const root = createRoot(container);
+  const params = new URLSearchParams(location.search.slice(1));
+  const cartoMapId = params.has('id') ? params.get('id')! : 'f45022b8-298b-44d5-8168-3ae329808914';
+
+  fetchMap({cartoMapId}).then(({initialViewState, layers}) => {
+    root.render(<App initialViewState={initialViewState} layers={layers} />);
+  });
+}
+
+function formatLabel(n: number, label: string) {
+  return `${label ? label + ': ' : ''}` + Math.round(n);
+}
+
+function ObjectSelect({title, obj, value, onSelect, top = 0}) {
+  const keys = Object.values(obj);
+  return (
+    <>
+      <select
+        onChange={e => onSelect(e.target.value)}
+        style={{position: 'absolute', padding: 4, margin: 2, width: 200, right: 0, top}}
+        value={value}
+      >
+        <option hidden>{title}</option>
+        {keys.map((f: any) => (
+          <option key={f} value={f}>
+            {`${title}: ${f}`}
+          </option>
+        ))}
+      </select>
+      <br></br>
+    </>
+  );
+}
+
+function formatCount(count: number): string {
+  if (count < 1000) {
+    return `${count}`;
+  }
+  if (count < 1000000) {
+    const thousands = Math.floor(count / 1000);
+    return `${thousands}k`;
+  }
+  const millions = Math.floor(count / 1000000);
+  return `${millions}M`;
+}
+
+function createParameters(blending) {
+  const parameters: any = {depthTest: false};
+  if (blending === 'additive') {
+    parameters.blendFunc = [GL.SRC_ALPHA, GL.DST_ALPHA];
+    parameters.blendEquation = GL.FUNC_ADD;
+  } else if (blending === 'subtractive') {
+    parameters.blendFunc = [GL.ONE, GL.ONE_MINUS_DST_COLOR, GL.SRC_ALPHA, GL.DST_ALPHA];
+    parameters.blendEquation = [GL.FUNC_SUBTRACT, GL.FUNC_ADD];
+  }
```

Comment:
Minor – let's use the approach from #8959 if possible!

Reply:
I'm not committing the test app, but I will use those in the future 👍